core/include/drivers/driver.h

```diff
@@ -0,0 +1,155 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/**
+ * @copyright 2018 NXP
+ *
+ * @file    driver.h
+ *
+ * @brief   Definition of the generic driver object opened, closed,
+ *          suspended and resumed by the system.
+ *
+ */
+#ifndef __DRIVER_H__
+#define __DRIVER_H__
+
+#include <scattered_array.h>
+#include <tee_api_types.h>
+#include <trace.h>
+
+#if TRACE_LEVEL >= TRACE_DEBUG
+#define DRV_MGT_DEBUG
+#endif
+
+#ifndef CFG_PLAT_PWR
+/**
+ * @brief   Enumerate the driver power mode state
+ */
+enum drv_pwrmode {
```

Comment:
I feel this should be limited to STATE_RUNNING, STATE_SUSPENDED.
I think a drvier that is OFF was simply not probed and will not be used, so no need to store its state.

Reply:
I don't see any problem with the default state because the platfrom can add more state. In complex system, driver can not be always ON/SUSPEND. In some system, there is intermediate state like:
- Clock Gating
- Idle that is not a OFF nor a suspend because driver (clock/power) is not completely cut
- A driver can be OFF in case system is going to OFF and so the driver must save some HW configuration that mut be restored at wakeup.

Reply:
I think that if the system goes OFF, the driver will not wakeup but will be probed again and initialized through the normal init sequence.

Reply:
By experience, system OFF is the deepest low power mode that can be reached but this is not the switch (ON/OFF) or remove power supply. This is a mode where system can still be wakeup by an external interrupt. 
If you have an other term, you're welcome.